backend/Controllers/user_controller.py

```diff
@@ -205,12 +208,28 @@ def get_user(request, user_id):
              401 - login required.
              404 - user not found.
     """
+
     try:
         get_user = User.objects.get(id=user_id)
     except:
         return JsonResponse(
             "Not Found - User does not exist.", status=404, safe=False
         )
+
+    # Check for share code.
+    valid_sc = False
+    if get_user.share_code:
+        if request.GET.get("sharecode") == get_user.share_code:
+            valid_sc = True
+
+    if not valid_sc:
+        try:
```

Comment:
This doesn't work 100%; I've tested it and when logged in, you can get the profile even if the code is invalid (or not given, same thing essentially).

Reply:
We agreed that any logged in user should be able to see another user's profile. If you dont have an account/dont want to make one, then you would need the share code.

Reply:
Oh right I see. All good then